set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java

```diff
@@ -534,6 +546,41 @@ void validateIsDefinedBy(UserSet webUserSet) throws ParamValidationException, Re
 	}
     }
 
+	/**
+	 * validated the EntityBestItemsSet
+	 *
+	 * @param webUserSet
+	 * @throws ParamValidationException
+	 * @throws RequestBodyValidationException
+	 */
+	void validateEntityBestItemsSet(UserSet webUserSet) throws ParamValidationException, RequestBodyValidationException {
+		if (!isEntityBestItemsSet(webUserSet)) {
+			return;
+		}
+
+		// subject must be present
+		if (webUserSet.getSubject() == null) {
+			throw new ParamValidationException(UserSetI18nConstants.USERSET_VALIDATION_MANDATORY_PROPERTY,
+					UserSetI18nConstants.USERSET_VALIDATION_MANDATORY_PROPERTY,
+					new String[] { WebUserSetModelFields.SUBJECT, String.valueOf(webUserSet.getSubject())});
+		}
+
+		if (webUserSet.isOpenSet()) {
+			throw new ParamValidationException(UserSetI18nConstants.USERSET_VALIDATION_PROPERTY_NOT_ALLOWED,
+					UserSetI18nConstants.USERSET_VALIDATION_PROPERTY_NOT_ALLOWED,
+					new String[] { WebUserSetModelFields.IS_DEFINED_BY, webUserSet.getType() });
+		}
+
+		if (!webUserSet.isEntityBestItemSet()) {
+			throw new RequestBodyValidationException(UserSetI18nConstants.USERSET_VALIDATION_ENTITY_REFERENCE,
```

Comment:
the validation rules (business logic) must be implemented here not in the model classes

Reply:
done